lib/utils.dart

```diff
@@ -70,42 +70,62 @@ Future<void> showAlertDialog(BuildContext context, String title, List<Widget> bo
 
 void showSharingSheet(BuildContext context, ShareLinks links) {
   var locals = AppLocalizations.of(context)!;
+  bool isAddTimestamp = false;
```

Comment:
the problem here is that the add timestamp option is going to show when sharing channel url or a video that is not playing (from the standard video screen) as well, which obviously is not something we want form a user experience stand point. I'd probably add an optional bool parameter to this method to show or not the checkbox.

Also to get the current timestamp of the video player, you'll probably need to following code or something similar:
```
Duration? timestamp = await        PlayerController.to()?.videoController?.videoPlayerController?.position;
```

Reply:
Added a boolean `showShareWithTimestamp` that defaults to false. Is that sufficient?

Reply:
yes, but to nitpick a bit there are 2 places that the share video button appears. First when user displays a video (before playing it) where we should be able to share the timestamp because this video is not playing yet in theory and another share button in the  "player" where i believe should be the only place where this timestamp thingy is showing.

Reply:
Aah yes, thankyou! Sorry if this PR became longer than coding this yourself :sweat_smile:

Reply:
nah it's fine. I'm glad people are willing to help :)

Reply:
@lamarios quick question - if I watch a video and see it in the playlist later, the player remembers the time I watched it till. Where is this done/stored? 

The reason to ask is - instead of enabling this feature only on the miniplayer, I think it would be nice to have it when you share a video preview and it's already watched. I think it should be easy with the info I asked above, but if you think it's not worth it we can add timestamps only via video player. Wdyt?

Reply:
It's in the local database, see global `db` object but i think it's over-engineering. Let's keep it simple. if someone's watching the video and wants to share it as they watch offer them to share the timestamp otherwise it's not necessary.

Reply:
Sharing with timestamp will only be shown from miniplayer :)